plugins/ledger/config/initializers/ledger/client.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+#
+# basic ledger event client
+#
+require 'faraday'
+require 'openssl'
+
+LEDGER_URI = "/api/v1/events"
+
+module Ledger
+  class Client
+    class << self
+      def verify_env
+        raise "LEDGER_TOKEN is not set in ENV" unless ENV.fetch("LEDGER_TOKEN", false)
+        raise "LEDGER_BASE_URL is not set in ENV" unless ENV.fetch("LEDGER_BASE_URL", false)
+      end
+
+      def post_deployment(deploy)
+        verify_env
+        deployment(deploy)
+      end
+
+      private
+
+      def create_event(event)
+        events = { "events": [event] }
+        results = post(events)
+        # TODO: raise or display errors if results.status != 200
+        results
+      end
+
+      def deployment(deploy)
```

Comment:
methods that do something should have a verb in their name ... post_deployment or send_deployment